### ISSUE TITLE
`no_prox` Flag

### ISSUE THREAD

**Issue by Zach-ER** [sail-sg/Adan#12]
Hi there,

I'm just wondering about the `no_prox` setting.

First of all, does it stand for "approximation"?

In the paper, Algorithm 1, line 7 corresponds to `no_prox=True` —
why is the default setting in this repo `False`? Why do you include this option at all?

Were the experiments in the paper done as the algorithm states, or with `no_prox=True`?

Again, I really appreciate the work! Am just struggling with this detail.

**Comment by Zach-ER** [comment]
Ah ok you're right, I was mistaken. I agree that L7 corresponds to `no_prox=False`. 

I don't understand what you mean by: 

> The final result does not fluctuate greatly. Hence, we choose a more responsible way to decay weight.

Do you just mean that you think your method is more principled? If so, why do you include the option of turning it off (maybe just to replicate your experiments?)

**Comment by XingyuXie** [comment]
1. We prefer our new way (`no_prox=False`) to perform weight decay since it is more responsible from the optimization perspective. It is not because it has better experimental results.
2. Someone may prefer the old way (`no_prox=True`) to perform weight decay since this way is more prevalent in ViT/ConvNext training at present. For this reason, we provide this option.  
3. All results for Adan are obtained by setting `no_prox=False`. If you have no special preference for the type of weight decay. You may ignore the option for `no_prox=True`.

**Comment by Zach-ER** [comment]
OK, thanks for your input! 

Could I just ask one more thing? What does `prox` stand for here? 🙈

**Comment by XingyuXie** [comment]
`prox` stands for the proximal operator, which is the key idea to obtaining our way of weight decay (`no_prox=False`). if `no_prox=True`, then we degenerate to the way used in AdamW.
